packages/core/src/odata-common/entity-builder.spec.ts

```diff
@@ -38,40 +40,145 @@ describe('EntityBuilder', () => {
     ).toEqual(expected);
   });
 
-  it('should build an entity from json', () => {
-    const builder = TestEntity.builder();
-    const linkedEntity = TestEntitySingleLink.builder()
-      .stringProperty('someString')
-      .withCustomFields({ linkedCustomField: 'someLinkedValue' })
-      .build();
-
-    const entityJson: Partial<TestEntity> = {
-      stringProperty: 'someValue',
-      toSingleLink: linkedEntity
-    };
-    entityJson['_customFields'] = { additionalField: 'someAdditionalValue' };
-    const actual = builder.fromJson(entityJson);
-    expect(actual.stringProperty).toBe(entityJson.stringProperty);
-    expect(actual.getCustomFields()).toEqual(entityJson['_customFields']);
-    expect(actual.toSingleLink.stringProperty).toBe(
-      linkedEntity.stringProperty
-    );
-    expect(actual.toSingleLink.getCustomFields()).toEqual(
-      linkedEntity.getCustomFields()
-    );
-  });
-
   it('should build an entity with non-primitive JS types (moment, BigNumber etc.)', () => {
     const expected: TestEntity = new TestEntity();
 
-    expected.dateTimeProperty = moment();
+    expected.dateProperty = moment();
     expected.decimalProperty = new BigNumber(10);
 
     const actual = TestEntity.builder()
-      .dateTimeProperty(expected.dateTimeProperty)
+      .dateProperty(expected.dateProperty)
       .decimalProperty(expected.decimalProperty)
       .build();
 
     expect(expected).toEqual(actual);
   });
+
+  describe('fromJson', () => {
+    it('should build an entity from json', () => {
```

Comment:
Could you perhaps also add one E2E test with the custom fields for the `GET` and `CREATE/UPDATE` case so that we see, that serialization/desrialization works end to end.

Reply:
Not necessary, as serializers are not called for custom fields.